src/main/java/com/scaler/lld/design/assignments/prototype/ConfigurationPrototypeRegistryImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.scaler.lld.design.assignments.prototype;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConfigurationPrototypeRegistryImpl implements ConfigurationPrototypeRegistry{
+    Map<ConfigurationType,Configuration> registry=new HashMap<>();
+    @Override
+    public void addPrototype(Configuration user) {
+        registry.put(user.getType(),user);
+    }
+
+    @Override
+    public Configuration getPrototype(ConfigurationType type) {
+        return registry.get(type);
+    }
+
+    @Override
+    public Configuration clone(ConfigurationType type) {
+        Configuration cf=registry.get(type);
```

Comment:
1. An edge case you should handle here is the prototype object being null or not present in the registry. Try to use the `Optional` class instead of an if-else case.
2. [Minor] Always try to create variable names which are readable. `configuration` > `conf` > `cf`.